src/AutoRest.CSharp/Common/Input/Source/SourceInputModel.cs

```diff
@@ -56,6 +64,15 @@ public ModelTypeMapping CreateForModel(INamedTypeSymbol? symbol)
             return new ModelTypeMapping(_modelAttribute, _schemaMemberNameAttribute, symbol);
         }
 
+        public IMethodSymbol? FindProtocolMethod(string namespaceName, string clientName, string methodName, IEnumerable<string> parameters)
+        {
+            if (_existingCompilation is ProtocolCompilationInput protocolCompilation)
+            {
+                return protocolCompilation.FindSymbol(namespaceName, clientName, methodName, parameters);
+            }
+            return null;
+        }
```

Comment:
Because `SourceInputModel` is used in multiple target, a method that only finds "protocol" methods that only exist in DPG does not make sense.
We should generalize this method as `FindMethod`. From the parameter list of this method, there is no clue that shows these parameters can only map to a protocol method. If I pass in namespace, client name, method name and parameters of an existing other type of method, I should also get it.

Reply:
Also please add some comments on this method to explain the purpose of each parameters.
And I believe we should not use `clientName`, instead we should use `typeName`, to indicate that this method is "finding a method on a type with the specific signature".

Reply:
What does the `IEnumerable<string> parameters` mean? Is this a list of parameter names or parameter types?
Ideally, this should be a `IEnumerable<IParameterSymbol> parameters` or `IEnumerable<ITypeSymbol>` or `IEnumerable<CSharpType>`. If this means a list of parameter type names (which is a string array), it should also work fine. But I believe they should be fully qualified names (namespace included to disambiguate as much as possible)

Reply:
We cannot do this. Different method would have different arguments.

Reply:
Synced to move this to base abstract class.